common/display/src/main/scala/uk/ac/wellcome/display/models/DisplayAccessCondition.scala

```diff
@@ -0,0 +1,53 @@
+package uk.ac.wellcome.display.models
+
+import java.time.LocalDate
+import io.circe.generic.extras.JsonKey
+import io.swagger.v3.oas.annotations.media.Schema
+import uk.ac.wellcome.models.work.internal.{AccessCondition, AccessStatus}
+
+@Schema(
+  name = "AccessCondition"
+)
+case class DisplayAccessCondition(
+  status: DisplayAccessStatus,
+  terms: Option[String],
+  to: Option[LocalDate],
+  @JsonKey("type") @Schema(name = "type") ontologyType: String =
+    "AccessCondition"
+)
+
+object DisplayAccessCondition {
+
+  def apply(accessCondition: AccessCondition): DisplayAccessCondition =
+    DisplayAccessCondition(
+      status = DisplayAccessStatus(accessCondition.status),
+      terms = accessCondition.terms,
+      to = accessCondition.to,
+    )
+}
+
+@Schema(
+  name = "AccessStatus"
+)
+case class DisplayAccessStatus(
+  id: String,
+  label: String,
+  @JsonKey("type") @Schema(name = "type") ontologyType: String = "AccessStatus"
+)
+
+object DisplayAccessStatus {
+
+  def apply(accessStatus: AccessStatus): DisplayAccessStatus =
+    accessStatus match {
+      case AccessStatus.Open =>
+        DisplayAccessStatus("open", "Open")
+      case AccessStatus.OpenWithAdvisory =>
+        DisplayAccessStatus("open-with-advisory", "Open with Advisory")
+      case AccessStatus.Restricted =>
+        DisplayAccessStatus("restricted", "Restricted")
+      case AccessStatus.Closed =>
+        DisplayAccessStatus("closed", "Closed")
+      case AccessStatus.LicensedResources =>
+        DisplayAccessStatus("licensed-resources", "Licensed Resources")
```

Comment:
Is keeping this information in the *display* model the right place for it?

I would lean towards putting it all on the core model, and reading those fields out here, a la License: https://github.com/wellcometrust/catalogue/blob/385c9bf87cd33edcd9ab2b8ee9d8e721bb2f8637/common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/License.scala

Reply:
We did have a discussion about this with @warrd. This is closer to how [notes are implemented](https://github.com/wellcometrust/catalogue/blob/master/common/display/src/main/scala/uk/ac/wellcome/display/models/DisplayNote.scala#L49). @warrd and I agreed to go in this direction rather because it makes it easier to change dispplay labels, ie we don't need a reindex. 
Do you have any particular concerns about it?

Reply:
(If I remember correctly, we also discussed migrating Licenses, WorkType etc to use this model. The idea is to push display stuff into the displaymodels as much as possible)

Reply:
Yeh as @alicefuzier points out this is connected to other discussions we have been having regarding models vs display models.

A big match statement with lots of case clauses isn't the prettiest / most declaritive way to represent the data, but in my opinion things like labels should gradually be removed from the internal model, and should just be part of the display. As mentioned it allows anything related to display to be modified independently to the internal model. Also, there are areas where logic would be much simpler if we had a "minimal" internal model, rather than storing lots of data on it (such as complications in the aggregations code).

It feels like quite a big discussion to be had within this, so maybe worth having a chat about things with everyone at some point.

Reply:
One problem with this approach is that, if it's only in the display models, it's not in es, so it's not as easy to search/aggregate. I tried to do something similar for WorkTypes and realised it would be a pain with aggregation, so worth keeping that in mind

Reply:
fwiw I'm very much in favour of moving labels to the display models from the perspective of search: the Terms aggregations that we have to use in order to fetch both the ID and label have been a bit of a pain point in (eg) https://github.com/wellcometrust/catalogue/pull/321

Reply:
If we think in terms of multi serialisation / deserialisation, we could have:

```scala
// I have used `ApiThing` to represent something similar to `ApiDisplayThing`
case class Work(workType: WorkType)
case class ApiDisplayWork(workType: ApiDisplayWorkType)
case class ElasticWork(workType: ElasticWorkType)

trait WorkType
case object Books extends WorkType
case object Pictures extends WorkType

case class ApiDisplayWorkType(label: String, id: String)
case object ApiDisplayWork {
  def serialize(workType: WorkType): ApiDisplayWorkType = {
    workType match {
      case Books => ApiDisplayWorkType("Books", "a")
      case Pictures => ApiDisplayWorkType("Pictures", "p")
    }
  }
}

case class ElasticWorkType(`type`: String) {
  def serialize(workType: WorkType): ElasticWorkType = {
    workType match {
      case Books => ElasticWorkType("books")
      case Pictures => ElasticWorkType("pictures")
    }
  }

  def deserialize(elasticWorkType: ElasticWorkType): WorkType = elasticWorkType.`type` match {
    case "books" => Books
    case "pictures" => Pictures
  }
}
```

There's other implementation of this approach discussed [here](https://github.com/wellcomecollection/stacks-service/pull/49#issuecomment-558740047) and [here](https://github.com/wellcometrust/catalogue/pull/311/files#r356175213) (sorry to keep ringing the same bell but actually though it was relevant).

I've left the implementation a little shoddy to get the gist across.

Reply:
Moving this [here](https://github.com/wellcometrust/platform/issues/3930#issuecomment-571501887) as it bikeshedding here.